### DISCUSSION TITLE
Many lettuce object in heap, can not GC

### DISCUSSION THREAD

**Discussion by yanyi5496** [redis/lettuce#2084] [Q&A]
lettuce version: 5.0.4.RELEASE
Application run two days, then I found it failed to work, this is log
![image](https://user-images.githubusercontent.com/48555554/164955267-b8fb5c1e-1d4e-4270-8911-87ea80efaabc.png)
![image](https://user-images.githubusercontent.com/48555554/164955277-f092df25-bc17-4dd1-acdb-daeac8f9c1eb.png)
![image](https://user-images.githubusercontent.com/48555554/164955310-3a84d518-e9b1-409c-85cb-a5c515656e07.png)
![image](https://user-images.githubusercontent.com/48555554/164955335-f327a6d5-927c-438d-a900-6b38cdc4cde5.png)
I use 'jmap -histo' show java heap, see this picture
![image](https://user-images.githubusercontent.com/48555554/164954974-485da2d2-dd12-4fbb-9563-8c7204ac153e.png)

redis and my application all in docker, How can I troubleshoot this， please provide me with some ideas. Thank you very much

**Comment by mp911de** [comment]
It seems that you have many open connections. You might want to make sure that you close connections after their usage. Also, having a report over which components hold on to connections makes sense to find potential leaks. 

Please upgrade to Lettuce 6 at your earliest convenience because Lettuce version 5 is no longer maintained.
